xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/test/java/org/xwiki/model/internal/reference/DefaultReferenceEntityReferenceResolverTest.java

```diff
@@ -312,12 +313,27 @@ public void resolveSpaceReferenceWhenTypeIsPage()
         EntityReference reference =
             this.resolver.resolve(new EntityReference("page", EntityType.PAGE), EntityType.SPACE);
 
-        assertEquals(new SpaceReference(DEFAULT_WIKI, Arrays.asList("page")), reference);
+        assertEquals(new SpaceReference(DEFAULT_WIKI, List.of("page")), reference);
 
         reference = this.resolver.resolve(
             new EntityReference("page1", EntityType.PAGE, new EntityReference("page2", EntityType.PAGE)),
             EntityType.SPACE);
 
-        assertEquals(new SpaceReference(DEFAULT_WIKI, Arrays.asList("page2", "page1")), reference);
+        assertEquals(new SpaceReference(DEFAULT_WIKI, List.of("page2", "page1")), reference);
+    }
+
+    @Test
+    void resolveRelativeEntityReference()
+    {
+        // When the space reference has a parent type parameter of tye space, then it's resolved using the base
+        // reference space has parent of it.
+        EntityReference relativeSpaceReference = new EntityReference("Space", EntityType.SPACE,
+            Map.of(EntityReference.PARENT_TYPE_PARAMETER, EntityType.SPACE));
+
+        EntityReference reference = this.resolver.resolve(relativeSpaceReference, EntityType.SPACE);
+        SpaceReference spaceReference = new SpaceReference(DEFAULT_WIKI, List.of(DEFAULT_SPACE, "Space"));
+        EntityReference expectedReference =
+            new EntityReference(spaceReference, Map.of(EntityReference.PARENT_TYPE_PARAMETER, EntityType.SPACE));
```

Comment:
wdyt, does it make sense to you that when resolving we keep the parameter? I'm not sure about that one.

Reply:
A reference which have a parent should definitely not have this parameter set.

Reply:
ok